spec/codecs/json_lines_spec.rb

```diff
@@ -117,6 +117,22 @@
       end
     end
 
+    describe "decode_size_limits_bytes" do
+      let(:maximum_payload) { "a" * subject.decode_size_limit_bytes }
+
+      it "should not raise an error if the number of bytes is not exceeded" do
+        expect {
+          subject.decode(maximum_payload)
+        }.not_to raise_error
+      end
+      
+      it "should raise an error if the max bytes are exceeded" do
+        expect {
+          subject.decode(maximum_payload << "z")
+        }.to raise_error(RuntimeError, "input buffer full")
```

Comment:
Unfortunately that exception is provided by the FileWatch library, so it'd require a patch there. I should probably wrap and re-raise it.